src/Analyzers/CSharp/Analyzers/NamespaceFileSync/NamespaceFileSyncDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,177 @@
+            // but now could possibly add an Error severity as an analyzer
+            var isCurrentNamespaceInvalid = namespaceDeclaration.Name
+                .GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error);
+            if (isCurrentNamespaceInvalid)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool ContainsPartialTypeWithMultipleDeclarations(NamespaceDeclarationSyntax namespaceDeclaration, SemanticModel semanticModel)
+        {
+            var partialMemberDecls = namespaceDeclaration
+                .Members.OfType<TypeDeclarationSyntax>()
+                .Where(t => t.Modifiers.Any(SyntaxKind.PartialKeyword));
+
+            foreach (var memberDecl in partialMemberDecls)
+            {
+                var memberSymbol = semanticModel.GetDeclaredSymbol(memberDecl);
+
+                // Simplify the check by assuming no multiple partial declarations in one document
+                if (memberSymbol is ITypeSymbol typeSymbol && typeSymbol.DeclaringSyntaxReferences.Length > 1)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool IsFileAndNamespaceValid(
+            NamespaceDeclarationSyntax namespaceDeclaration,
+            string rootNamespace,
+            string projectDir,
+            [NotNullWhen(returnValue: true)] out string? targetNamespace)
+        {
+            var filePath = namespaceDeclaration.SyntaxTree.FilePath;
+            if (!filePath.Contains(projectDir))
+            {
+                // The file does not exist within the project directory
+                targetNamespace = null;
+                return false;
+            }
+
+            var relativeFilePath = filePath.Substring(projectDir.Length);
+            var folders = Path.ChangeExtension(relativeFilePath, null).Split(Path.DirectorySeparatorChar).Where(e => e.Length > 0);
+            var expectedNamespace = PathMetadataUtilities.TryBuildNamespaceFromFolders(folders, CSharpSyntaxFacts.Instance, rootNamespace);
+
+            if (expectedNamespace is null || expectedNamespace.Equals(namespaceDeclaration.Name.ToString()))
+            {
+                // The namespace currently matches the folder structure
+                targetNamespace = null;
+                return false;
+            }
+
+            targetNamespace = expectedNamespace;
+            return true;
+        }
+
+        private static void ReportDiagnostics(
+           SyntaxNodeAnalysisContext context, DiagnosticDescriptor descriptor,
+           NamespaceDeclarationSyntax namespaceDeclaration, ReportDiagnostic reportDiagnostic,
+           string targetNamespace)
+        {
+            var properties = new Dictionary<string, string>()
+                { { "TargetNamespace", targetNamespace } }
+                .ToImmutableDictionary();
```

Comment:
`ImmutableDictionary.Create("TargetNamespace", targetNamespace);`

Reply:
That doesn't exist. The closest I could find to shorthand was `ImmutableDictionary<string, string>.Empty.Add("TargetNamespace", targetNamespace)` 

Is it an extension method?

Reply:
Ping

Reply:
interesting :)  sorry, i must have been mistaken.  do teh .Empty.Add approach though :)  Or ImmutableDictionary-Builder.